mailbox/backup/src/main/java/org/apache/james/mailbox/backup/InternalDateExtraField.java

```diff
@@ -0,0 +1,72 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.mailbox.backup;
+
+import java.time.Instant;
+import java.time.ZoneId;
+import java.time.ZonedDateTime;
+import java.util.Date;
+import java.util.Optional;
+
+import org.apache.commons.compress.archivers.zip.ZipShort;
+
+public class InternalDateExtraField extends LongExtraField {
+
+    public static final ZipShort ID = new ZipShort(0x6F61); // "ao" in little-endian
+
+    public InternalDateExtraField() {
+        super();
+    }
+
+    public InternalDateExtraField(Optional<Date> date) {
+        super(date
+            .map(Date::toInstant)
+            .map(instant -> ZonedDateTime.ofInstant(instant, ZoneId.systemDefault()))
+            .map(locateZonedDateTime -> locateZonedDateTime.withZoneSameInstant(ZoneId.of("UTC")))
+            .map(ZonedDateTime::toInstant)
+            .map(Instant::toEpochMilli));
+    }
+
+    public InternalDateExtraField(Date date) {
+        this(Optional.of(date));
+    }
+
+    public InternalDateExtraField(long timestamp) {
+        this(Optional.of(new Date(timestamp)));
+    }
+
+    @Override
+    public ZipShort getHeaderId() {
+        return ID;
+    }
+
+    public Optional<Date> getDateValue() {
+        return getValue().map(Date::new);
+    }
+
+    public Optional<Date> getLocalDateValue() {
+        return getValue()
+            .map(Instant::ofEpochMilli)
+            .map(instant -> ZonedDateTime.ofInstant(instant, ZoneId.of("UTC")))
+            .map(utcTime -> utcTime.withZoneSameInstant(ZoneId.systemDefault()))
+            .map(ZonedDateTime::toInstant)
+            .map(Date::from);
```

Comment:
but not here...

Reply:
See above explanation please. Your magical "getTime" misses UTC convertion...